pkg/controller/master/upgrade/node_controller.go

```diff
@@ -64,11 +63,19 @@ func (h *nodeHandler) OnChanged(key string, node *corev1.Node) (*corev1.Node, er
 	if !ok {
 		return node, nil
 	}
-	machine, err := h.machineClient.Get(rancherMachineNamespace, machineName, metav1.GetOptions{})
-	if err != nil {
-		return node, err
+
+	// find machine plan secret
+	secrets, err := h.secretClient.List(rancherPlanSecretNamespace, metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", rancherPlanSecretMachineLabel, machineName),
+		FieldSelector: fmt.Sprintf("type=%s", rancherPlanSecretType),
+	})
+
+	if err != nil || len(secrets.Items) != 1 {
+		return node, nil
```

Comment:
Ditto. Return `err`?

```suggestion
	if err != nil || len(secrets.Items) != 1 {
		return node, err
```